### PR TITLE
FIO checkTransactions\processTransaction functionality for history. FIO address registration API endpoint.

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: src/fio/fioPlugin.js [hunk 1]
```diff
@@ -140,10 +137,10 @@ export function makeFioPlugin(opts: EdgeCorePluginOptions): EdgeCurrencyPlugin {
       )
       return fioSDK.getPublicAddress(fioAddress, chainCode, tokenCode)
     },
-    validateAccount: async (fioAddress: string): boolean => {
+    validateAccount: async (fioAddress: string): Promise<boolean> => {
       if (
         !new RegExp(
-          `^(([a-z0-9]+)(-?[a-z0-9]+)*@{1}${currencyInfo.defaultSettings.fioDomain}{1})$`,
+          '^(?:(?=.{3,64}$)[a-zA-Z0-9]{1}(?:(?!-{2,}))[a-zA-Z0-9-]*[a-zA-Z0-9]+@[a-zA-Z0-9]{1}(?:(?!-{2,}))[a-zA-Z0-9-]*[a-zA-Z0-9]+$)',
           'gim'
```
Comment (swansontec) [inline src/fio/fioPlugin.js]: Who is `gim`?

File: src/fio/fioInfo.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ import { type EdgeCurrencyInfo } from 'edge-core-js/types'
 
 const defaultSettings: any = {
   apiUrls: ['https://testnet.fioprotocol.io:443/v1/'],
+  historyNodeUrls: ['https://testnet.fio.dev/v1/'],
```
Comment (swansontec) [inline src/fio/fioInfo.js:8]: Testnet?

File: src/fio/fioPlugin.js (hunk 1 above)
Comment (swansontec) [reply src/fio/fioPlugin.js]: I see you fix it later.

**Comment by andreyvEze** [comment]
> I think we should squash these commits, since they make more sense together as one.

2cfb1eb and 94ec731 ?
maybe we can squash all of it?

**Pull request merged**